contracts/data/WitnetBoardData.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "../exports/WitnetTypes.sol";
+
+/**
+ * @title Witnet Board basal data model. 
+ * @author Witnet Foundation
+ */
+abstract contract WitnetBoardData {  
+
+  struct SWitnetBoardData {
+    address owner;
+    address instance;
+    mapping (uint => SWitnetBoardDataRequest) requests;
+    uint256 noRequests;
+  }
+
+  struct SWitnetBoardDataRequest {
+    WitnetTypes.DataRequest dr;
+    bytes result;
+  }
+
+  constructor() {
+    __data().owner = msg.sender;
+  }
+
+  modifier onlyOwner {
+    require(msg.sender == __data().owner, "WitnetBoardData: only owner");
+    _;    
+  }
+
+  modifier resultNotYetReported(uint256 id) {
+    require(__dataRequest(id).txhash == 0, "WitnetBoardData: already solved");
+    _;
+  }
+
+  modifier wasPosted(uint256 id) {
+    require(id > 0 && id <= __data().noRequests, "WitnetBoardData: not yet posted");
+    _;
+  }
+
+  function __data() internal pure returns (SWitnetBoardData storage _struct) {
```

Comment:
```suggestion
  function _data() internal pure returns (SWitnetBoardData storage _struct) {
```

Afaik, the widely used convention is only 1 underscore for internal functions.
Still, it's nothing official after reviewing the [style guide](https://docs.soliditylang.org/en/latest/style-guide.html) from the solidity documentation.

For example, in the `ERC20` there is an underscore for internal functions, e.g. `_approve`, `_transfer`, ...

Reply:
I'd rather prefer to used double underscores for internal functions, as to avoid possible confusion when calling libraries on passed arguments (which are usually prefixed with one single underscore). We do agree however that internal functions should be prefixed w/ at least one underscore :-)